src/main/java/io/github/fablabsmc/fablabs/api/fiber/v1/builder/ConfigLeafBuilder.java

```diff
@@ -167,10 +167,12 @@ private ConfigLeafBuilder(ConfigTreeBuilder parentNode, @Nonnull String name, @N
 	 */
 	@Override
 	public ConfigLeaf<T> build() {
-		if (this.defaultValue != null) {
-			if (!this.type.accepts(this.defaultValue)) {
-				throw new RuntimeFiberException("Default value '" + this.defaultValue + "' does not satisfy constraints on type " + this.type);
-			}
+		if (this.defaultValue == null) {
```

Comment:
I think we should take a default value in the builder's constructor instead of checking at the end.